cdap-watchdog/src/main/java/co/cask/cdap/logging/save/CheckpointManager.java

```diff
@@ -37,50 +41,84 @@
   private static final byte [] OFFSET_COLNAME = Bytes.toBytes("nextOffset");
   private static final byte [] MAX_TIME_COLNAME = Bytes.toBytes("maxEventTime");
 
-  private final Transactional<DatasetContext<Table>, Table> mds;
   private final byte [] rowKeyPrefix;
+  private final LogSaverTableUtil tableUtil;
+  private final TransactionExecutorFactory transactionExecutorFactory;
+  private Map<Integer, Checkpoint> lastCheckpoint;
 
   public CheckpointManager(final LogSaverTableUtil tableUtil,
                            TransactionExecutorFactory txExecutorFactory, String topic, int prefix) {
     this.rowKeyPrefix = Bytes.add(Bytes.toBytes(prefix), Bytes.toBytes(topic));
-    this.mds = Transactional.of(txExecutorFactory, new Supplier<DatasetContext<Table>>() {
+    this.tableUtil = tableUtil;
+    this.transactionExecutorFactory = txExecutorFactory;
+    this.lastCheckpoint = new HashMap<>();
+  }
+
+  private <T> T execute(TransactionExecutor.Function<Table, T> func) {
+    try {
+      Table table = tableUtil.getMetaTable();
+      if (table instanceof TransactionAware) {
+        TransactionExecutor txExecutor = Transactions.createTransactionExecutor(transactionExecutorFactory,
+                                                                                (TransactionAware) table);
+        return txExecutor.execute(func, table);
+      } else {
+        throw new RuntimeException(String.format("Table %s is not TransactionAware, " +
+                                                   "Exception while trying to cast it to TransactionAware. " +
+                                                   "Please check why the table is not TransactionAware", table));
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(String.format("Error accessing %s table", tableUtil.getMetaTableName()), e);
+    }
+  }
+
+  public void saveCheckpoint(final Map<Integer, Checkpoint> checkpointMap) throws Exception {
+    LOG.trace("Saving checkpoints for partitions {}", checkpointMap);
```

Comment:
It would be a good idea to move this log message after the condition below, so that we log message only if there is a write.